src/riak_kv_qry_buffers.erl

```diff
@@ -0,0 +1,815 @@
+                      'ORDER BY' = OrderBy})
+  when OrderBy /= undefined ->
+    Part0 = erlang:phash2(From),
+    Part1 = erlang:phash2(ColNames),
+    Part2 = erlang:phash2(lists:sort(Where)),
+    Part3 = erlang:phash2(OrderBy),
+    {ok,
+     <<0,       %% marks an organic qref (1 = random qref for one-shot queries)
+       Part0:32/integer,
+       Part1:32/integer,
+       Part2:32/integer,
+       Part3:32/integer>>};
+make_qref(_) ->
+    {error, query_non_pageable}.
+
+
+-record(qbuf, {
+          %% original SELECT query this buffer holds the data for
+          orig_qry :: ?SQL_SELECT{},
+          %% a DDL for it
+          ddl :: ?DDL{},
+          %% table the original query selected data from
+          mother_table :: binary(),
+
+          %% this qbuf expiry period (set to a default value from
+          %% State; can be overridden)
+          expire_msec :: non_neg_integer(),
+
+          %% leveldb handle for the temp storage
+          ldb_ref :: eleveldb:db_ref(),
+
+          %% received chunks count so far
+          chunks_got = 0 :: integer(),
+          %% total chunks needed (== number of subqueries)
+          chunks_need :: integer(),
+
+          %% flipped to true when chunks_need == chunks_got
+          is_ready = false :: boolean(),
+
+          %% total records stored (when complete)
+          total_records = 0 :: non_neg_integer(),
+
+          %% %% iterator cache (need a working iterator support in eleveldb)
+          %% iter_cache = [] :: [cached_iter()],
+
+          %% total size on disk, for reporting
+          size = 0 :: non_neg_integer(),
+
+          %% last added chunk (when not complete) or queried
+          last_accessed = 0 :: erlang:timestamp(),
+
+          %% a proplist for all your options
+          options = [] :: qbuf_options(),
+
+          %% precomputed key field positions
+          key_field_positions :: [non_neg_integer()],
+
+          %% process waiting for this qbuf ready status
+          ready_waiting_pid :: pid() | undefined
+         }).
+
+-record(state, {
+          qbufs = [] :: [{qbuf_ref(), #qbuf{}}],
+          total_size = 0 :: non_neg_integer(),
+          %% no new queries; accumulation allowed
+          soft_watermark :: non_neg_integer(),
+          %% drop some tables now
+          hard_watermark :: non_neg_integer(),
+          %% drop incomplete query buffer after this long since last add_chunk
+          incomplete_qbuf_release_msec :: non_neg_integer(),
+          %% drop complete query buffers after this long since serving last query
+          qbuf_expire_msec :: non_neg_integer(),
+          %% dir containing qbuf ldb files
+          root_path :: string()
+         }).
+
+
+-spec start_link() -> {ok, pid()} | ignore | {error, term()}.
+start_link() ->
+    gen_server:start_link({local, ?MODULE}, ?MODULE, [], []).
+
+
+-spec init(proplists:proplist()) -> {ok, #state{}}.
+%% @private
+init(Options) ->
+    RootPath =
+        filename:join(
```

Comment:
I do feel like there are usability benefits for a configurable path. A user might want to store this data on AWS instance storage. Configurable paths would prevent symlink hell for users that want to take advantage.  A riak.conf default of `${platform_data_dir}/query_buffers` would align nicely with AAE data path configurations (https://github.com/basho/riak_kv/blob/develop/priv/riak_kv.schema#L123-L128).   Might be good to get an additional CSE eyeball on this.